catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/SourceOperations.java

```diff
@@ -394,18 +389,21 @@ private SourceInfoResponse getFanoutSourceInfo(SourceInfoRequest sourceInfoReque
           String sourceId = source.getId();
           LOGGER.debug("adding sourceId: {}", sourceId);
 
-          Source cachedSource = null;
-          // check the poller for cached information
-          if (frameworkProperties.getSourcePoller() != null
-              && frameworkProperties.getSourcePoller().getCachedSource(source) != null) {
-            cachedSource = frameworkProperties.getSourcePoller().getCachedSource(source);
+          final Set<ContentType> cachedContentTypes;
```

Comment:
❓Would this work instead?
```suggestion
          final Set<ContentType> cachedContentTypes =
            contentTypesCache.getCachedValueForSource(source)
            .orElseGet(() -> {
              LOGGER.debug("Unknown ContentTypes for source id={}", sourceId);
              return Collections.emptySet();
            });
```

Reply:
See https://github.com/codice/ddf/pull/3858#discussion_r253677955

Reply:
See https://github.com/codice/ddf/pull/3858/files#r253930138 😉